examples/regexp.fx

```diff
@@ -0,0 +1,87 @@
+import Re2
+
+fun example(funname: string, arg1: 'a, arg2: 'b, f: ('a, 'b) -> 't): void
+{
+	println(f"Re2.{funname}(\"{arg1}\", r\"{arg2}\") = {f(arg1, arg2)}")
+}
+
+fun example(funname: string, arg1: 'a, arg2: 'b, arg3: 'c, f: ('a, 'b, 'c) -> 't): void
+{
+	println(f"Re2.{funname}(\"{arg1}\", \"{arg2}\", r\"{arg3}\") = {f(arg1,arg2,arg3)}")
+}
+
+fun example(funname: string, arg1: 'a, arg2: 'b, arg3: 'c, arg4: 'd, arg5: 'e, f: ('a, 'b, 'c, 'd, 'e) -> 't): void
+{
+	println(f"Re2.{funname}(\"{arg1}\", \"{arg2}\", \"{arg3}\", \"{arg4}\", \"{arg5}\") = {f(arg1,arg2,arg3,arg4,arg5)}")
+}
+
+println("Instant regexp matching")
+example("full_match", "hello", r"h.*o", Re2.full_match) // Bugreport1: Why in this case we CAN OMIT signature.
+example("full_match", "hello", r"e", Re2.full_match)
+example("partial_match", "hello", r"h.*o", Re2.partial_match)
+example("partial_match", "hello", r"e", Re2.partial_match)
+
+println()
+println("Matching with predefined regexps")
+
+val reg_ho = Re2.compile(r"h.*o")
+val reg_e  = Re2.compile(r"e")
+
+example("full_match", "hello", reg_ho, (Re2.full_match: (string, Re2.regex_t)->bool))   // Bugreport1: why in this case we MUST define signature directy 
```

Comment:
because the first overloaded variant of Re2.full_match is the incorrect match here, but the compiler tries to use it anyway

Reply:
Ok.